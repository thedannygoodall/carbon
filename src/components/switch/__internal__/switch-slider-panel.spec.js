import React from "react";
import TestRenderer from "react-test-renderer";
import { css } from "styled-components";

import {
  assertStyleMatch,
  carbonThemesJestTable,
} from "../../../__spec_helper__/test-utils";
import StyledLoader from "../../loader/loader.style";
import StyledLoaderSquare from "../../loader/loader-square.style";
import SwitchSliderPanel from "./switch-slider-panel.style";

function render(props) {
  return TestRenderer.create(<SwitchSliderPanel {...props} />);
}

describe("SwitchSliderPanel", () => {
  describe("base theme", () => {
    it("renders as expected", () => {
      expect(render()).toMatchSnapshot();
    });

    describe("when loading === true", () => {
      it("applies the correct Loader styles", () => {
        assertStyleMatch(
          {
            width: "100%",
            height: "100%",
          },
          render({ isLoading: true }).toJSON(),
          {
            modifier: css`
              ${StyledLoader}
            `,
          }
        );
      });

      it("applies the correct LoaderSquare styles", () => {
        assertStyleMatch(
          {
            width: "var(--sizing200)",
            height: "var(--sizing200)",
          },
          render({ isLoading: true, size: "large" }).toJSON(),
          {
            modifier: css`
              ${StyledLoader} ${StyledLoaderSquare}
            `,
          }
        );
      });
    });
  });

  describe.each(carbonThemesJestTable)(
    "when the theme is set to %s",
    (theme) => {
      const wrapper = render({ theme }).toJSON();

      it("applies the correct base styles", () => {
        assertStyleMatch(
          {
            color: "var(--colorsActionMinorYang100)",
          },
          wrapper
        );
      });

      it("applies the correct off panel styles", () => {
        assertStyleMatch(
          {
            color: "var(--colorsActionMinor500)",
          },
          wrapper,
          { modifier: '[type="off"]' }
        );
      });
    }
  );
});
