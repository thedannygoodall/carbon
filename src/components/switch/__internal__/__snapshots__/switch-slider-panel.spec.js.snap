// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchSliderPanel base theme renders as expected 1`] = `
.c0 {
  border: 0;
  color: var(--colorsActionMinorYang100);
  margin: auto;
  margin-top: 5px;
}

.c0[type="on"] {
  margin-left: 9px;
}

.c0[type="off"] {
  color: var(--colorsActionMinor500);
  margin-right: 6px;
}

<div
  className="c0"
/>
`;
